### ISSUE TITLE
Interfaces are removed from lambda groups when shrinking is disabled

### ISSUE THREAD

**Issue by heckej** [heckej/proguard#10]
When shrinking is disabled, all classes and methods are marked as used, excluding the lambda groups. The methods that call methods of the lambda groups, will cause those lambda group methods to be marked as used. Afterwards, the unused lambda group methods are shrunk away. However, apparently also the interfaces are removed, which causes cast exceptions when the enclosing class tries to cast the lambda group to some `kotlin.jvm.functions.Function` interface.

**Comment by heckej** [comment]
The `ClassUsageMarker` does not mark the constants from the lambda group constant pool that are related to its interfaces. As a result, the interface constants are removed, and therefore the interfaces array is emptied. Perhaps, adding a `SignatureAttribute` to the lambda group could solve this problem.

**Comment by heckej** [comment]
> adding a `SignatureAttribute`

does not solve the problem. However, explicitly marking the interfaces of the lambda groups as used before shrinking ensures that the interface constants are kept:
```
SimpleUsageMarker simpleUsageMarker = new SimpleUsageMarker();
ClassUsageMarker classUsageMarker = new ClassUsageMarker(simpleUsageMarker);
lambdaGroupClassPool.classesAccept(new InterfaceUsageMarker(classUsageMarker));
```

**Comment by heckej** [comment]
Fixed by commit [ebf4d8d2d2e8e1b0f66a532666cdb5598c69bdca](https://github.com/heckej/proguard/commit/ebf4d8d2d2e8e1b0f66a532666cdb5598c69bdca)

**Comment by heckej** [comment]
TODO: write a test for this fix
